### PR TITLE
Feat/mb 2 dropdown

### PR DESCRIPTION
**Pull request by sarkin271**
## What
Fixed typos
Included font awesome library
Styled dropdown dropdown

## Why
We don't want no typos in our project
Submenu required a caret icon
Styling dropdown now matches designs and does not mess up the navigation styling

## How
Changed text in HTML file
Added in Font Awesome library, then added in appropriate caret icon for the list items
Updated the css that directly effected the submenu and dropdown

### CODE REVIEW

**Comment by jarkin13** [comment]
@sarkin271  - good job with the PR description, it's a good improvement. 

# Pull Request Description
I would like the description to be more high level of what you accomplished. Here is a good article detailing a good PR description - https://www.pullrequest.com/blog/writing-a-great-pull-request-description/. While we are doing a basic dev project, it's good for you to get some practice with this. To add markup to your description please review: https://guides.github.com/features/mastering-markdown/

In addition, you should use your PR description to use "human language" rather than the specifics of the code you implemented. For instance, 

Changed position for carrots and dropdown menu to absolute and positioned appropriately  

Could be something like:

Added in appropriate FA caret icon to each menu item

I like their template:

## What?
## Why?
## How?
## Testing? (not applicable at this point)
## Screenshots (optional)
## Anything Else?

# Commits
Each commit should accomplish one thing. For example, the first commit adding in font awesome should be:
- Adding in the font awesome library
- Linking to the FA library in `index.html`
- Adding in icons from FA ex: `<i class="fas fa-caret-down"></i>` in index.html
- Personally, all styling would go into a different commit, unless the styling ONLY involved css for the icon

Then the commit would be something like "add in icons for dropdown menu items". Again, it should be human language to sum up what you have done.

**Comment by jarkin13** [comment]
@sarkin271 reorganize the assets:
- move `webfonts` to the `font-awesome` folder
- create new folder in `v1` called `assets`
- move `images` and `font-awesome` folders to`v1/assets`
- update all URLs to correct path

File: v1/index.html [hunk 1]
```diff
@@ -10,6 +10,7 @@
       rel="stylesheet"
     />
     <link rel="shortcut icon" href="images/mybookie-tablogo.jpg" />
+    <link rel="stylesheet" href="../font-awesome/all.css" />
```
Comment (jarkin13) [inline v1/index.html]: after finishing my reorganizing assets comment - update to `v1/assets/font-awesome/all.css`

File: v1/index.html [hunk 2]
```diff
@@ -49,7 +50,7 @@
       </nav>
       <div class="nav nav-secondary-container">
         <div class="container">
-          <div class="row align-items-center">
+          <div class="row align-items-center relative-container">
```
Comment (jarkin13) [inline v1/index.html]: why is this a row with `position: relative`? If you want the div to be positioned relative, remove `.row`

File: v1/styles.css [hunk 3]
```diff
@@ -270,8 +271,26 @@ ul {
   background-color: var(--grey100);
 }
 
+.relative-container {
```
Comment (jarkin13) [inline v1/styles.css]: remove

File: v1/styles.css [hunk 4]
```diff
@@ -270,8 +271,26 @@ ul {
...
+  top: 2.5rem;
+  padding: 0;
+  width: 120px;
+}
+
+.btn-link__submenu li {
+  padding: 8px 16px;
+}
+
+.fas {
```
Comment (jarkin13) [inline v1/styles.css]: don't style `.fas` directly, create a new class called `.icon` or `.fa-icon` whatever you'd like

File: v1/styles.css [hunk 5]
```diff
@@ -222,6 +222,7 @@ ul {
   height: 32px;
   display: flex;
   align-items: center;
+  margin-left: 8px;
```
Comment (jarkin13) [inline v1/styles.css]: what does this have to do with adding in font awesome?

File: v1/styles.css [hunk 6]
```diff
@@ -270,10 +271,14 @@ ul {
...
 
+.relative-container {
+  position: relative;
+}
+
 .btn-link__submenu.opened {
   display: block;
   position: absolute;
-  top: 19%;
+  top: 2.5rem;
```
Comment (jarkin13) [inline v1/styles.css:279]: why is this in the commit for adding in font awesome?

File: v1/styles.css [hunk 7]
```diff
@@ -407,6 +426,12 @@ ul {
   padding: 0;
 }
 
+.section-header__item-right {
+  text-align: right;
+  padding-right: 16px;
+  color: var(--grey100);
```
Comment (jarkin13) [inline v1/styles.css:438]: why is `color` included in `item-right`?

File: v1/styles.css [hunk 8]
```diff
@@ -407,6 +426,12 @@ ul {
   padding: 0;
 }
 
+.section-header__item-right {
+  text-align: right;
```
Comment (jarkin13) [inline v1/styles.css:436]: a var should be set in `section-header__item` for lines 430 & 431. Update variable in `right` instead.

File: v1/styles.css [hunk 9]
```diff
@@ -407,6 +426,12 @@ ul {
   padding: 0;
 }
 
+.section-header__item-right {
```
Comment (jarkin13) [inline v1/styles.css]: update to `section-header_item--right` - please review http://getbem.com/introduction/

**Review by jarkin13 — CHANGES_REQUESTED** [review]
Please fix the following (I made comments detailing the changes I want made):
- [ ] Pull request description
- [ ] Code I made comments on
- [ ] Reorganize assets

File: v1/index.html (hunk 2 above)
Comment (sarkin271) [reply v1/index.html]: I tried removing row and relative container, but it didn't look right

File: v1/styles.css (hunk 7 above)
Comment (sarkin271) [reply v1/styles.css:438]: I included it in item-right, because the color is different from the left item, but I don't have a specific class for the left item.

File: v1/styles.css (hunk 8 above)
Comment (sarkin271) [reply v1/styles.css:436]: I don't understand what you mean.

File: v1/styles.css (hunk 5 above)
Comment (sarkin271) [reply v1/styles.css]: Nothing I guess, was just trying to fix the spacing of the right carrot, because it was too close to help center.

File: v1/styles.css (hunk 6 above)
Comment (sarkin271) [reply v1/styles.css:279]: This has nothing to do with font awesome I suppose, I was fixing the positioning of the carrots and this is when I decided to add the relative container for both.

**Comment by sarkin271** [comment]
Did it like this because that's how they had it in the website, but doesn't look great. If you want me to condense them all under one what, why and how let me know.

**Comment by sarkin271** [comment]
Also closed this by accident and don't know how to undo.

## What
Changed "odds form" to "odds format"
Added container for position relative containing odds format, time zone, and help center
Added font awesome to project and referenced in the head and called carrots
Changed position for carrots and dropdown menu to absolute
Styled dropdown menu with padding
Made small correction to content header spacing

_here is what I would write,_
- Fixed typos
- Included font awesome library
- Styled dropdown dropdown

 ## Why
"odds form" was a typo
Changed to position relative so when positioning the dropdown menu and carrots I didn't have to position according to the whole page, because when I did when the screen size changed the positioning didn't stay consistent.
Needed font-awesome as a reference for the carrots
Needed to fix positioning
Wanted dropdown menu to look similar to actual mybookie
Color and spacing of right item wasn't right

_here is what I would write,_
- We don't want no typos in our project 
- Submenu required a caret icon 
- Styling dropdown now matches designs and does not mess up the navigation styling

## How
Changed text in HTML file
Added class of relative-container to div
Downloaded and added filed to project and called next to odds form/time zone
Did so in their classes in css
Changed padding in their css classes
Added new class to the right item

_here is what I would write,_
- Changed text in HTML file
- Added in Font Awesome library, then added in appropriate caret icon for the list items 
- Updated the css that directly effected the submenu and dropdown

File: v1/index.html (hunk 2 above)
Comment (jarkin13) [reply v1/index.html]: Well, then debug why it didn't look right, don't just add this class because you don't know why it didn't look right.

File: v1/styles.css (hunk 5 above)
Comment (jarkin13) [reply v1/styles.css]: Moving forward items like this should be in a separate commit.

File: v1/styles.css (hunk 6 above)
Comment (jarkin13) [reply v1/styles.css:279]: Again, moving forward move this to a different commit.

File: v1/styles.css (hunk 7 above)
Comment (jarkin13) [reply v1/styles.css:438]: Yeah I understand that you want the text to be grey. This styling should not be in class called `item-right`. Only styling that makes the item right should go into this class. Not having a class named `item-left` is irrelevant to what styling goes into `item-left`. 

Please fix.

**Comment by jarkin13** [comment]
@sarkin271 I don't see where they add multiple **headers** (ex: why, how, etc). If you read the article, in the different sections they are discussing different examples of how to use what, how and why - but not telling you to have multiple headers. 

Here are two examples they provided:

# Example 1
## What?
I've added support for authentication to implement Key Result 2 of OKR1. It includes model, table, controller and test. For more background, see ticket 
#JIRA-123.
## Why?
These changes complete the user login and account creation experience. See #JIRA-123 for more information.
## How?
This includes a migration, model and controller for user authentication. I'm using Devise to do the heavy lifting. I ran Devise migrations and those are included here.
## Testing?
I've added coverage for testing all new methods. I used Faker for a few random user emails and names.
## Screenshots (optional)
0
## Anything Else?
Let's consider using a 3rd party authentication provider for this, to offload MFA and other considerations as they arise and as the privacy landscape evolves. AWS Cognito is a good option, so is Firebase. I'm happy to start researching this path. Let's also consider breaking this out into its own service. We can then re-use it or share the accounts with other apps in the future.

--------------
# Example 2
## What?
I've added support for authentication to implement Key Result 2 of OKR1. It includes model, table, controller and test. For more background, see ticket #JIRA-123.
## Why?
These changes complete the user login and account creation experience. See #JIRA-123 for more information.
## How?
This includes a migration, model and controller for user authentication. I'm using Devise to do the heavy lifting. I ran Devise migrations and those are included here.
## Testing?
I've added coverage for testing all new methods. I used Faker for a few random user emails and names.

File: v1/index.html (hunk 2 above)
Comment (sarkin271) [reply v1/index.html]: I'll look for other ways to make this work without the new class, but I still don't understand why it's bad for this row to have position relative.

File: v1/styles.css (hunk 7 above)
Comment (sarkin271) [reply v1/styles.css:438]: I don't understand what you want from me here. How else could I style just the right element without adding a new class?

**Comment by jarkin13** [comment]
@sarkin271 can you update the description with the new description you posted? Also, not entirely sure what you mean by "Also closed this by accident and don't know how to undo."

In addition, I still feel as if your descriptions are too specific, too technical and not "human language". For example:
**Downloaded and added filed to project and called next to odds form/time zone** means little to me if I just started the review. Rewriting this to **Added the Font Awesome library and included the caret markup provided** makes a lot more sense.

**Comment by sarkin271** [comment]
I see your descriptions are much simpler haha I'll try to be more concise next time. I accidentally pressed close with comment and a.) I don't really know what that means and b.) I don't know how to undo said closing

File: v1/index.html (hunk 2 above)
Comment (sarkin271) [reply v1/index.html]: I added position: relative to .row and it works is that better?

Comment (sarkin271) [reply v1/index.html]: Although you said to remove .row so I'm not sure thats how you want it.

Comment (jarkin13) [reply v1/index.html]: I want you to take out `.row` `.align-items-center` and `.relative-container`. I want you then to figure out what styling elements from `.row` and `.align-items-center` that make it look good. I then want you to create a new class with specific styling for the nav. 

I don't want you to just create a class that will only be used once and that overrides one of the main styling features `.row` applies, which is `.flex`.

File: v1/styles.css (hunk 7 above)
Comment (jarkin13) [reply v1/styles.css:438]: I dont want `color: var(--grey100)` in a generic reusable class. Again if the styling isn't being reused it should have a class specific to that item. Having a class change a div's text color, that is supposed to ONLY MOVE THE ITEM RIGHT is quite confusing. Either rename it to specific class name to that item or move the color to something else like `.text-grey300` and add that class in the HTML. 

RULE OF THUMB - _Stop overthinking stuff_. Only add a generic class name if that specific styling will get reused. Otherwise name it something specific to that html element.

**Comment by jarkin13** [comment]
@sarkin271 Thank you for updating the description. I know I provided you with exactly what I would do, but moving forward I do that to give you an example to make it your own. So if I give you something, in order for you to practice this you should read my example and add your own words, you make want to add or remove things.

File: v1/styles.css (hunk 7 above)
Comment (sarkin271) [reply v1/styles.css:438]: If I understand correctly that is what I was trying to do, I named it item--right because it was the item on the right. It has the text-align: right because it was on the left side of the right half it that makes sense. I could also change it to .section-header__item > div if you don't like the class name would that be better?
